File: models/contrato.py
"""Contrato — valor fixo R$ 2.000,00, parcelável em até 5 vezes (RF10)."""


class Contrato:
    VALOR_CONTRATO = None  # deve valer 2000.00 na fase GREEN

    def __init__(self, qtd_parcelas: int):
        pass

    def calcular_valor_parcela(self) -> float:
        raise NotImplementedError

    def get_qtd_parcelas(self) -> int:
        raise NotImplementedError

File: src/models/Contrato.ts
/** Contrato — valor fixo R$ 2.000,00, parcelável em até 5 vezes (RF10). */
import { arredondar } from "../utils/arredondamento.js";
import { MAX_PARCELAS_CONTRATO, VALOR_CONTRATO } from "../utils/constants.js";

export class Contrato {
  static readonly VALOR_CONTRATO = VALOR_CONTRATO;

  private qtdParcelas: number;

  constructor(params: { qtdParcelas: number }) {
    const { qtdParcelas } = params;
    if (qtdParcelas < 1 || qtdParcelas > MAX_PARCELAS_CONTRATO) {
      throw new Error(
        `Quantidade de parcelas deve ser entre 1 e ${MAX_PARCELAS_CONTRATO}.`,
      );
    }
    this.qtdParcelas = qtdParcelas;
  }

  calcularValorParcela(): number {
    return arredondar(Contrato.VALOR_CONTRATO / this.qtdParcelas);
  }

  getQtdParcelas(): number {
    return this.qtdParcelas;
  }
}
